jupyterlab_git/git.py

```diff
@@ -872,13 +874,50 @@ async def init(self, current_path):
         Execute git init command & return the result.
         """
         cmd = ["git", "init"]
+        cwd = os.path.join(self.root_dir, current_path)
         code, _, error = await execute(
-            cmd, cwd=os.path.join(self.root_dir, current_path)
+            cmd, cwd=cwd
         )
 
+        actions = None
+        if code == 0:
+            code, actions = await self._maybe_run_actions('post_init', cwd)
+
         if code != 0:
-            return {"code": code, "command": " ".join(cmd), "message": error}
-        return {"code": code}
+            return {"code": code, "command": " ".join(cmd), "message": error, "actions": actions}
+        return {"code": code, "actions": actions}
+
+    async def _maybe_run_actions(self, name, cwd):
+        code = 0
+        actions = None
+        if self._config and name in self._config.actions:
+            actions = []
+            actions_list = self._config.actions[name]
+            for action in actions_list:
+                try:
+                    # We trust the actions as they were passed via a config and not the UI
+                    code, stdout, stderr = await execute(
+                        shlex.split(action), cwd=cwd
+                    )
+                    actions.append({
+                        'cmd': action,
+                        'code': code,
+                        'stdout': stdout,
+                        'stderr': stderr
+                    })
+                    # After any failure, stop
+                except Exception as e:
+                    code = 1
+                    action.append({
```

Comment:
```suggestion
                    actions.append({
```

This is obviously not tested 😜

Reply:
Good catch!